src/main/java/io/github/techstreet/dfscript/script/values/ScriptBoolValue.java

```diff
@@ -18,7 +21,23 @@ String typeName() {
     }
 
     @Override
-    public String asText() {
+    public ScriptValue convertTo(ScriptValue type) {
+        if (type instanceof ScriptDictionaryValue) {
+            HashMap<String, ScriptValue> map = new HashMap<>();
+            map.put("Boolean", this);
+            return new ScriptDictionaryValue(map);
+        } else if (type instanceof ScriptListValue) {
+            List<ScriptValue> list = List.of(this);
+            return new ScriptListValue(list);
+        } else if (type instanceof ScriptNumberValue) {
+            return new ScriptNumberValue(asNumber());
+        } else {
+            return super.convertTo(type);
+        }
+    }
```

Comment:
what is this change? we adding bool casting at the same time?

Reply:
sorry this is my first proper pr got a little carried away